src/compass/s1_geocode_metadata.py

```diff
@@ -236,16 +236,43 @@ def geocode_luts(geo_burst_h5, burst, cfg, dst_group_path, item_dict,
             isce3.io.Raster(
                 f"IH5:::ID={dst_dataset.id.id}".encode("utf-8"), update=True)
 
-        # populate and prepare radargrid LUT input raster
+        # Define the radargrid for LUT interpolation
+        radargrid_interp = radar_grid.copy()
+
+        radargrid_interp.width = int(np.ceil(burst.width / dec_factor))
+        intv_interp_range = (burst.width - 1) / (radargrid_interp.width - 1)
+        range_px_interp_vec = np.arange(0, burst.width, intv_interp_range)
+
+        radargrid_interp.length = int(np.ceil(burst.length / dec_factor))
+        intv_interp_azimuth = (burst.length - 1) / (radargrid_interp.length - 1)
+        azimuth_px_interp_vec = np.arange(0, burst.length, intv_interp_azimuth)
+
+        radargrid_interp.range_pixel_spacing *= intv_interp_range
+        radargrid_interp.prf /= intv_interp_azimuth
```

Comment:
I wanted to ensure that the very first and the last elements of the LUT be included in the new radargrid. In that case, `multilook()` needs to take `azlooks` and `rglooks` as floating point, rather than integer. So I decided to calculate the new radargrid based on `dec_factor`, which is nominal decimation factor.

I've added a comment to explain why I directly calculated the radargrid, Otherwise even I would be perplexed by myself when I revisit this code after few months ;)

Reply:
I think a multilooked radargrid generated by will cover the first and last elements of the LUT. When multilooked, the sensing and slant range start will [change](https://github.com/isce-framework/isce3/blob/944eba17f4a5b1c88c6a035c2d58ddd0d4f0709c/cxx/isce3/product/RadarGridParameters.h#L173) but grid size in azimuth (seconds) and slant range (meters) will remain the same since the multilooked PRF and range pixel spacing have been scaled accordingly. This is because the sensing and slant range starts are the center of the first pixel in their respective direction and not the actual leading edge of the pixel. The computations in [`isce3::product::RadarGridParameters::contains()`](https://github.com/isce-framework/isce3/blob/944eba17f4a5b1c88c6a035c2d58ddd0d4f0709c/cxx/isce3/product/RadarGridParameters.cpp#L28) demonstrate that.

I've updated my suggestion above to capture additional commands needed to compute the LUTs.